src/server/service/GameService.ts

```diff
@@ -0,0 +1,229 @@
+        }
+        catch {
+            this.checkGameFinishOrEnd(createdRoom);
+        }
+        return createdRoom;
+    }
+
+    public addUserToRoom(playerIndex: string, indexRoom: string) {
+        const currentPlayer = this.playerRepository.getPlayerByIndex(playerIndex);
+        if (!currentPlayer) throw new Error('Player not found');
+        
+        this.roomRepository.addUserToRoom(indexRoom, currentPlayer);
+    }
+
+    public updateRoom(roomIndex: string): Room {
+        const room = this.roomRepository.getRoomByIndex(roomIndex);
+        if (!room) throw new Error('Room not found');
+
+        if (room.players.length >= 2) {
+            return room;
+        } else if (room.players.length === 0) {
+            this.roomRepository.removeRoom(room);
+        }
+
+        throw new Error('Room state undefined');
+    }
+
+    public getRoom(roomIndex: string): Room | undefined {
+        return this.roomRepository.getRoomByIndex(roomIndex);
+    }
+
+    public getFreeRooms(): Room[] {
+        return this.roomRepository.getAllRooms().filter(room => room.players.length < 2);
+    }
+
+    public addShipsToRoom(playerIndex: string, shipsData: AddShipsRequest): boolean {
+        const room = this.roomRepository.getRoomByIndex(shipsData.gameId);
+        if (!room || room.isStarted) throw new Error('Room not found or battle already started');
+
+        const ships: Ship[] = shipsData.ships.map(ship => ({
+            position: ship.position,
+            direction: ship.direction,
+            length: ship.length,
+            type: ship.type,
+            hits: 0
+        }));
+        room.ships.set(playerIndex, ships);
+
+        return room.ships.size >= 2;
+    }
+
+    public startGame(roomIndex: string): Room {
+        const room = this.roomRepository.getRoomByIndex(roomIndex);
+        if (!room || room.isStarted) throw new Error('Room not found or battle already started');
+
+        room.isStarted = true;
+        this.switchToNextPlayer(room);
+        return room;
+    }
+
+    private executeAttack(playerIndex: string, roomIndex: string, x: number, y: number): AttackResult[] {
+        const attackResults: AttackResult[] = [];
+
+        const room = this.roomRepository.getRoomByIndex(roomIndex);
+        if (!room || !room.isStarted) throw new Error('Invalid room or game not started');
+        if (room.nextPlayerIndex !== playerIndex) throw new Error('Invalid player turn');
+
+        const opponentIndex = room.players.find(player => player.index !== playerIndex)?.index;
+        const opponentShips = room.ships.get(opponentIndex!);
+        if (!opponentShips) throw new Error('Opponent ships not found');
+
+        const hitShip = opponentShips.find(ship => this.isHit(ship, x, y));
+        if (!hitShip) {
+            attackResults.push(new AttackResult(x, y, 'miss'));
+            this.switchToNextPlayer(room);
+            return attackResults;
+        } else {
+            hitShip.hits++;
```

Comment:
As I mentioned in the previous comment, you do not consider if this position has been hit previously.

Reply:
Here is the video clip of the bug

https://github.com/user-attachments/assets/f8c16684-77a0-4466-a259-5318cf975323